c_src/exml.cpp

```diff
@@ -480,6 +481,13 @@ static ERL_NIF_TERM parse_next(ErlNifEnv *env, int argc,
   if (!parser->copy_buffer(env, argv[1]))
     return enif_make_badarg(env);
 
+  // Raise an exception when null character is found.
+  const char *data = reinterpret_cast<const char*>(&Parser::buffer[0]);
+  if (std::strlen(data) != Parser::buffer.size() - 1)
```

Comment:
While it is correct, the problem is that `std:strlen` is an O(n) operation. Decoding xml is already O(n) so this would mean we're scanning the string twice. I'd prefer here introduce the check on the scan that is already happening, I introduced something similar here: https://github.com/esl/exml/pull/62 (I'm wondering if this bug now is somehow related to those last changes of mine 🤔 )